includes/apple-exporter/builders/class-components.php

```diff
@@ -293,6 +302,67 @@ private function anchor_together( $component, $target_component ) {
 	}
 
 	/**
+	 * Add a thumbnail if needed.
+	 *
+	 * @param array &$components
+	 * @access private
+	 */
+	private function add_thumbnail_if_needed( &$components ) {
+		// If a thumbnail is already defined, just return.
+		if ( $this->content_cover() ) {
+			return;
+		}
+
+		// Otherwise, iterate over the components and look for the first image.
+		foreach ( $components as $i => $component ) {
+			if ( is_a( $component, 'Apple_Exporter\Components\Image' ) ) {
+				// Get the bundle URL of this class.
+				$json_url = $component->get_json( 'URL' );
+				if ( empty( $json_url ) ) {
+					$json_components = $component->get_json( 'components' );
+					if ( ! empty( $json_components[0]['URL'] ) ) {
+						$json_url = $json_components[0]['URL'];
+					}
+				}
+
+				if ( empty( $json_url ) ) {
+					return;
+				}
+
+				// Isolate the bundle URL basename
+				$bundle_basename = str_replace( 'bundle://', '', $json_url );
```

Comment:
To handle the extremely unlikely event that the substring occurs somewhere other than beginning of `$json_url`, maybe do `preg_replace( '/^bundle:\/\//', '', $json_url );`